src/logging/mod.rs

```diff
@@ -0,0 +1,162 @@
+    }
+}
+
+impl Value for Settings {
+    fn serialize(&self, _: &Record, _: Key, serializer: &mut Serializer) -> slog::Result {
+        serializer.emit_str("auth_db_uri", &self.authdb.baseuri)?;
+        serializer.emit_str("provider", &self.provider)?;
+
+        if let Some(_) = self.aws.keys {
+            serializer.emit_str("aws_keys_access", "[hidden]")?;
+        } else {
+            serializer.emit_str("aws_keys_access", "[not set]")?;
+        }
+
+        serializer.emit_str("aws_region", &self.aws.region)?;
+
+        if let Some(ref sqsurls) = self.aws.sqsurls {
+            serializer.emit_str("aws_sqs_urls_bounce", &sqsurls.bounce.to_owned())?;
+            serializer.emit_str("aws_sqs_urls_complaint", &sqsurls.complaint.to_owned())?;
+            serializer.emit_str("aws_sqs_urls_delivery", &sqsurls.delivery.to_owned())?;
+            serializer.emit_str("aws_sqs_urls_notification", &sqsurls.notification.to_owned())?;
+        } else {
+            serializer.emit_str("aws_sqs_urls", "[not set]")?;
+        }
+
+        if let Some(ref jsonlogging) = self.jsonlogging {
+            serializer.emit_str("json_logging", &format!("{}", jsonlogging))?;
+        }
+
+        if let Some(_) = self.sendgrid {
+            serializer.emit_str("sendgrid_key", "[hidden]")?;
+        } else {
+            serializer.emit_str("sendgrid_key", "[not set]")?;
+        }
+
+        serializer.emit_str("smtp_host", &self.smtp.host)?;
+        serializer.emit_str("smtp_port", &format!("{}", self.smtp.port))?;
+        
+        if let Some(ref user) = self.smtp.user {
+            serializer.emit_str("smtp_user", user)?;
+        } else {
+            serializer.emit_str("smtp_user", "[not set]")?;
+        }
+
+        if let Some(_) = self.smtp.password {
+            serializer.emit_str("smtp_password", "[hidden]")?;
+        } else {
+            serializer.emit_str("smtp_password", "[not set]")?;
+        }
+
+        serializer.emit_str("bounce_limits_enabled",  &format!("{}", &self.bouncelimits.enabled))?;
+        serializer.emit_str("bounce_limits_complaint",  &format!("{:?}", &self.bouncelimits.complaint))?;
+        serializer.emit_str("bounce_limits_hard",  &format!("{:?}", &self.bouncelimits.hard))?;
+        serializer.emit_str("bounce_limits_soft",  &format!("{:?}", &self.bouncelimits.soft))?;
+
+        Ok(())
+    }
+}
+
+pub struct MozlogLogger(slog::Logger);
+
+impl MozlogLogger {
+    pub fn with_request(request: &Request) -> Result<MozlogLogger> {
+        let logger = request
+            .guard::<State<MozlogLogger>>()
+            .success_or(err_msg("Internal error: No managed MozlogLogger"))?;
+        Ok(MozlogLogger(
+            logger.new(slog_o!(RequestMozlogFields::from_request(request))),
+        ))
+    }
+}
+
+impl Deref for MozlogLogger {
+    type Target = slog::Logger;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+pub fn init_logging(settings: &Settings) -> Result<MozlogLogger> {
```

Comment:
Can we make this a member of `MozlogLogger` and rename it to just `init`? I'm thinking it might be nicer to invoke it as `MozLogger::init` (or maybe `MozLogger::new`), since that associates it more strongly with the type its returning.